<launch>
    <!-- Global paramters -->
    <arg name="odom" default="/odom" />
    <arg name="ground_plane" default="/ground_plane" />
    <arg name="tf_target_frame" default="map" />
    <arg name="bayes_people_param_file" default="$(find bayes_people_tracker)/config/detectors.yaml" />
    <arg name="bayes_people_positions" default="/people_tracker/positions" />
    <arg name="bayes_people_pose" default="/people_tracker/pose" />
    <arg name="bayes_people_pose_array" default="/people_tracker/pose_array" />
    <arg name="bayes_people_people" default="/people_tracker/people" />
    <arg name="bayes_people_marker" default="/people_tracker/marker_array" />
    <arg name="scan" default="/scan_clf" />

    <arg name="machine" default="localhost" />
    <arg name="user" default="" />

    <machine name="$(arg machine)" address="$(arg machine)" env-loader="$(optenv ROS_ENV_LOADER )" user="$(arg user)" default="true"/>

    <!-- People Tracker -->
    <include file="$(find bayes_people_tracker)/launch/people_tracker.launch">
        <arg name="machine" value="$(arg machine)"/>
        <arg name="user" value="$(arg user)"/>
        <arg name="param_file" value="$(arg bayes_people_param_file)"/>
        <arg name="target_frame" value="$(arg tf_target_frame)"/>
        <arg name="positions" value="$(arg bayes_people_positions)"/>
        <arg name="pose" value="$(arg bayes_people_pose)"/>
        <arg name="pose_array" value="$(arg bayes_people_pose_array)"/>
        <arg name="people" value="$(arg bayes_people_people)"/>
        <arg name="marker" value="$(arg bayes_people_marker)"/>
    </include>

    <!-- To PoseArray -->
    <include file="$(find detector_msg_to_pose_array)/launch/to_pose_array.launch">
        <arg name="machine" value="$(arg machine)"/>
        <arg name="user" value="$(arg user)"/>
    </include>

    <param name="leg_detector/fixed_frame" value="base_footprint" />
    <include file="$(find leg_detector)/launch/leg_detector.launch">
        <arg name="scan" value="$(arg scan)" />
    </include>

</launch> 
